PocketKit/Tests/PocketKitTests/SearchViewModelTests.swift

```diff
@@ -198,4 +243,118 @@ class SearchViewModelTests: XCTestCase {
 
         XCTAssertEqual(viewModel.recentSearches, ["search-term-3", "search-term-4", "search-term-5", "search-term-6", "search-term-2"])
     }
+
+    // MARK: - Saves Searches
+    func test_savesSearches_withFreeUser_showSearchResults_searchTitle() throws {
+        user.status = .free
+
+        let savedItems = (1...2).map {
+            space.buildSavedItem(
+                remoteID: "saved-item-\($0)",
+                createdAt: Date(timeIntervalSince1970: TimeInterval($0))
+            )
+        }
+        try space.save()
+
+        source.stubSearchItems { _ in return savedItems }
+
+        let viewModel = subject()
+        var results = viewModel.searchItems(with: "saved")
+        XCTAssertTrue(results.count == 2)
+
+        source.stubSearchItems { _ in return [] }
+
+        results = viewModel.searchItems(with: "none")
+        XCTAssertTrue(results.isEmpty)
+
+        try space.clear()
+    }
+
+    func test_savesSearches_withPremiumUser_showSearchResults_searchTitle() throws {
+        user.status = .premium
+
+        let savedItems = (1...2).map {
+            space.buildSavedItem(
+                remoteID: "saved-item-\($0)",
+                createdAt: Date(timeIntervalSince1970: TimeInterval($0))
+            )
+        }
+        try space.save()
+
+        source.stubSearchItems { _ in return savedItems }
+
+        let viewModel = subject()
+        var results = viewModel.searchItems(with: "saved")
+        XCTAssertTrue(results.count == 2)
+
+        source.stubSearchItems { _ in return [] }
+
+        results = viewModel.searchItems(with: "none")
+        XCTAssertTrue(results.isEmpty)
+
+        try space.clear()
+    }
+
+    func test_savesSearches_withFreeUser_showSearchResults_searchTag() throws {
+        user.status = .free
+
+        _ = (1...2).map {
+            space.buildSavedItem(
+                remoteID: "saved-item-\($0)",
+                createdAt: Date(timeIntervalSince1970: TimeInterval($0)),
+                tags: ["test"]
+            )
+        }
+        try space.save()
+
+        source.stubSearchItems { _ in return [] }
+
+        let viewModel = subject()
+        var results = viewModel.searchItems(with: "saved")
```

Comment:
should this be "test" if we're trying to search on tags? also maybe rename test to shouldNotShowResults

Reply:
I think that testing Title and URL are the same.  They are both string comparisons that take place.  I don't see a benefit in testing both if one of them works as intended.
Tags are a bit different in that Premium searches search Tag whereas non-premium doesn't search tags.
So I'm testing the behavior of Core Data returning based on user status.

Reply:
thanks for the insight! that sounds good to me too, only suggestion is to modify `saved` to `test` and add the assertion

Reply:
> Title and URL aren't guaranteed to have the same strings in them.
i'm not saying that Title and URL would have the same strings, but if one string comparison works, the other should also work.
But understood, I'll add another test for URLs

Reply:
thanks for adding the url test, but seems like we didn't make the update from `saved` to `test` unless I'm missing something. also I like adding the assertion that @dskuza mentioned above 

> I think we can have an additional assertion (within this test) for testing against the tag "test" to verify that we are returning 0 items.

Reply:
yup, sorry. i think it got lost when i was fixing a bunch of merge conflicts.

Reply:
I still dont quite understand the reason for this.  Aren't I stubbing the searchItems to return an empty array?  Unless I'm just not understanding whats being asked.

Reply:
Oh I see the updated changes now! Looks good The reason is that we're checking specifically for tags, so would want to make sure we are checking `viewModel.searchItems(with: "test")`